app/charts/map/map-state.tsx

```diff
@@ -148,6 +152,61 @@ const useMapState = ({
   const getAreaValue = useOptionalNumericVariable(areaLayer.measureIri);
   const getSymbolValue = useOptionalNumericVariable(symbolLayer.measureIri);
 
+  const getDataByHierarchy = useCallback(
+    ({
+      geoDimensionIri,
+      hierarchyLevel,
+      getLabel,
+    }: {
+      geoDimensionIri: string;
+      hierarchyLevel: number;
+      getLabel: (d: Observation) => string;
+    }) => {
+      const dimension = dimensions.find((d) => d.iri === geoDimensionIri);
+
+      if (isGeoShapesDimension(dimension)) {
+        const hierarchyLabels = dimension.geoShapes.hierarchy
+          .filter((d: any) => d.level === hierarchyLevel)
+          .map((d: any) => d.label);
+
+        return data.filter((d) => hierarchyLabels.includes(getLabel(d)));
+      }
+
+      return data;
```

Comment:
Instead of having a callback, I think we could directly index the shapes by level so that we can use them directly afterwards without recomputation.

const dataByLevel = useMemo(() => { groupBy(data, x=> getLevelByLabel(x.label)) }